recipes/systemd_service.rb

```diff
@@ -42,6 +42,11 @@ class ::Chef::Recipe
 
 service_unit_content['Service'].delete('Restart') if timer
 
+if node['chef_client']['systemd']['timeout']
```

Comment:
based on documentation, it seems like this needs to check to make sure that timer isn't a oneshot and if it is do a WARN as this setting would be ignored in that case.